docs/legacy/developer/dependencies.md

```diff
@@ -2,35 +2,63 @@
-   * On NixOS with home-manager, you can set `programs.direnv.enable = true;`.
-   * Make sure direnv is hooked into your shell via it's appripriate `rc` file.
+1. Install [Nix](https://nixos.org/download.html). macOS users with a recent
+   Mac and non-NixOS Linux users should follow [these
+   instructions](https://nixos.org/manual/nix/stable/installation/installing-binary.html)
+   to install a binary distribution of Nix.
+
+   Debian users (running at least Debian 11) can use this distro's `nix-bin`
+   package. In this case, ensure your user is added to the `nix-users` group
+   (and restart your login session to pick this up). Note that Debian's Nix
+   package provides a *multi-user* installation (as opposed to a single-user
+   installation, which is an option when using the binary distribution tarball).
+2. Add the `wire-server` Cachix cache to your system. On NixOS, this can be
+   done by adding the following snippet to your system configuration:
+   ```nix
+   {
+     nix.binaryCachePublicKeys = [ "wire-server.cachix.org-1:fVWmRcvdsqzKek3X5Ad8nYNsBSjKZ9Um2NMLfMLS77Y=" ];
+     nix.binaryCaches = [ "https://wire-server.cachix.org" ];
+   }
+   ```
+   On non-NixOS systems, these need to be manually added to your `nix.conf`
+   (this is at `~/.config/nix/nix.conf` for single-user installations, and
+   `/etc/nix/nix.conf` for multi-user installations; you can create this file if
+   it doesn't exist):
+   ```
+   substituters = https://cache.nixos.org https://wire-server.cachix.org
+   trusted-public-keys = wire-server.cachix.org-1:fVWmRcvdsqzKek3X5Ad8nYNsBSjKZ9Um2NMLfMLS77Y= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+   ```
+3. Install [Direnv](https://direnv.net/).
+   On Debian, you can install the `direnv` package. On MacOS use `brew install direnv`.
+   * Make sure direnv is hooked into your shell via its appripriate `rc` file.
      Add `eval "$(direnv hook bash|zsh|fish)"` to your ~/.(bash|zsh|fish)rc .
-   * When successfully installed and hooked, direnv should ask you to `direnv allow`
-     the current `.envrc` when you cd to this repository.
-     See the [Installation documentation](https://direnv.net/docs/installation.html) for further details.
+
+   On NixOS with home-manager, you can set `programs.direnv.enable = true;`,
+   which does all of the above.
+
+   See the [Installation documentation](https://direnv.net/docs/installation.html)
+   for further details.
+
+   When successfully installed and hooked, direnv should ask you to `direnv allow`
+   the current `.envrc` when you cd to this repository.
+4. As this repo uses git submodules, ensure you're fetching from them automatically.
+   Run `git config --global submodule.recurse true` to enable this globally.
+   Run `git config --global fetch.parallel 10` to fetch more submodules in parallel.
+   If you've already cloned this repo before having enabled this, run `git
+   submodule update --init --recursive` once.
+
+#### Stack's Nix Integration
```

Comment:
Should this remain, if we're moving away from stack?

Reply:
I wanted to move that text somewhere, and not just delete it. Once we switch away from stack, I'd expect we purge all these docs as well.

As written in the PR description, there's also a bunch of other instructions on how to install various packages on various distros, I'm not sure if they still apply or if that's all covered by the dev env.

Ideally this should be cleaned up.